debug_toolbar/static/debug_toolbar/js/utils.js

```diff
@@ -75,7 +75,11 @@ function ajax(url, init) {
     return fetch(url, init)
         .then(function (response) {
             if (response.ok) {
-                return response.json();
+                    return response.json().catch(function(error){
+                        return Promise.reject(
+                            new Error("The response  is a invalid Json object : " + error)
+                        );
+                    });
```

Comment:
Yeah but if we have an hypotetical request as json header but return an invalid json it will not work.

Reply:
Yes, that's of course correct. I'm not sure if it's a big problem, but it certainly is an edge case.

Reply:
Judging from the [original issue here](https://github.com/jazzband/django-debug-toolbar/issues/1847#issuecomment-1775228325), it appears the user had put the djdt views behind `login_required` decorator or something similar. Since the responses were from `http://127.0.0.1:8000/login/?next=/__debug__/history_sidebar/` it's likely that it was a content type of `text/html`.

Given that @elineda, do you think it's more reasonable to check based on the content type rather than the body of the response?

Reply:
That's fair. @matthiask I don't feel strongly on this one way or another. What are your thoughts?

Reply:
I think the explanation makes sense; I think it would be fine to crash when the response headers do not match the content in this case, but I do not have anything against fixing this either even if it takes a little bit more code to do so.

Reply:
So it's ok like this ?

Reply:
As per our Djangonaut Space call, this is good to go.